indolaw-parser/parser_utils.py

```diff
@@ -318,56 +403,158 @@ def clean_law(law: List[str]) -> List[str]:
-    '''
-    law = clean_split_plaintext(law)
+    while next_cleaning_stage <= len_CleaningStageOrder:
+        
+        if next_cleaning_stage > 1:
 
-    save_law_to_file(law, 'e.txt')
+            print_line()
+            print("")
+            print(colored('SAVED STAGES: ', 'blue'))
+            for stage in CleaningStageOrder:
+                if len(CLEANING_STAGES[stage]['law_cleaned']) == 0:
+                    break
+                print(f"{stage.value}. {stage.name}")
+            print("")
+            print_line()
+            
+            pick_number = colored('number', 'blue')
+            pick_n = colored('n', 'blue')
 
-    '''
-    TODO(johnamadeo): Fix "Pasal 38 B" REMOVE THE SPACE WITH REGEX
-    '''
-    law = clean_split_pasal_number(law)
+            print(f"Pick a stage {pick_number} or")
+            input_stage = input(f"enter {pick_n} to go to the next stage: ")
 
-    save_law_to_file(law, 'f.txt')
+        if input_stage == 'n':
+            '''Next cleaning stage depends on last cleaned stage'''
+            print(f"next cleaning stage is {next_cleaning_stage}")
+            input_stage = next_cleaning_stage
 
-    '''
-    Add OPEN_QUOTE_CHAR and CLOSE_QUOTE_CHAR to PERUBAHAN_SECTION and PENJELASAN_PERUBAHAN_SECTION
-    '''
-    law = insert_perubahan_quotes(law)
-    return law
+        elif not input_stage.isnumeric():
+            print("")
+            print(f"{colored('Invalid input. Choose n or a number', 'red')}")
+            continue
+
+        int_input_stage = int(input_stage)
+
+        if int_input_stage > len_CleaningStageOrder:
+            print("")
+            print(f"{colored('Invalid number. No stage with that number.', 'red')}")
+            continue
 
+        elif (int_input_stage != 1 and len(CLEANING_STAGES[CleaningStageOrder(int_input_stage - 1)]['law_cleaned']) == 0):
+            '''
+            Checks whether or not the previous stage from int_input_stage has a valid law list
+            If not, then the stage of int_input_stage should not be implemented
+            and the terminal should re-ask the user for input
+            '''
+            
+            print("")
+            print(f"{colored('Invalid stage number. All former stages must be cleaned first.', 'red')}")
+            continue
+        
+        '''
+        Uses clean_law_at_stage to find which function to implement
+        Read documentation on clean_law_at_stage for more information
+        '''
+        if int_input_stage == 1:
+            new_law = clean_law_at_stage(int_input_stage, law)
+            CLEANING_STAGES[CleaningStageOrder(int_input_stage)]['law_cleaned'] = new_law
+            
+            next_cleaning_stage += 1
+
+        elif int_input_stage in range(2, len_CleaningStageOrder + 1):
+            try:
+                new_law = clean_law_at_stage(int_input_stage, CLEANING_STAGES[CleaningStageOrder(int_input_stage - 1)]['law_cleaned'])
```

Comment:
Instead of pulling out CleaningStageOrder(int_input_stage - 1) I took the liberty to change CLEANING_STAGE[CleaningStageOrder(int_pick_stage) into current_stage and CLEANING_STAGE[CleaningStageOrder(int_pick_stage - 1) into previous_stage. It seems to me like the code is still understandable, although with another layer of naming abstraction now haha. Do you agree?

*int_input_stage is also changed into int_pick_stage just to be consistent with other naming conventions.